x-pack/plugins/infra/public/pages/metrics/hosts/components/host_details_flyout/metadata/table.tsx

```diff
@@ -0,0 +1,115 @@
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EuiText, EuiFlexGroup, EuiFlexItem, EuiLink, EuiBasicTable } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { first } from 'lodash';
+import React, { useCallback, useMemo, useState } from 'react';
+import { FormattedMessage } from '@kbn/i18n-react';
+
+interface Row {
+  name: string;
+  value: string | string[] | undefined;
+}
+
+interface Props {
+  rows: Row[];
+}
+
+/**
+ * Columns translations
+ */
+const FIELD_LABEL = i18n.translate('xpack.infra.hostsViewPage.hostDetail.metadata.field', {
+  defaultMessage: 'Field',
+});
+
+const VALUE_LABEL = i18n.translate('xpack.infra.hostsViewPage.hostDetail.metadata.value', {
+  defaultMessage: 'Value',
+});
+
+export const Table = (props: Props) => {
+  const { rows } = props;
+  const columns = useMemo(
+    () => [
+      {
+        field: 'name',
+        name: FIELD_LABEL,
+        width: '35%',
+        sortable: false,
+        render: (name: string, item: Row) => <EuiText size="s">{name}</EuiText>,
+      },
+      {
+        field: 'value',
+        name: VALUE_LABEL,
+        width: '65%',
+        sortable: false,
+        render: (_name: string, item: Row) => {
+          return (
+            <>
+              {!Array.isArray(item.value) && item.value}
+              {Array.isArray(item.value) && <ArrayValue values={item.value} />}
+            </>
+          );
+        },
+      },
+    ],
+    []
+  );
+
+  return <EuiBasicTable tableLayout={'fixed'} responsive={false} columns={columns} items={rows} />;
+};
+
+interface MoreProps {
+  values: string[];
+}
+const ArrayValue = (props: MoreProps) => {
+  const { values } = props;
+  const [isExpanded, setIsExpanded] = useState(false);
+  const expand = useCallback(() => {
+    setIsExpanded(true);
+  }, []);
+
+  const collapse = useCallback(() => {
+    setIsExpanded(false);
+  }, []);
+
+  return (
+    <>
+      {!isExpanded && (
+        <EuiFlexGroup gutterSize={'xs'} responsive={false} alignItems={'baseline'} wrap={true}>
+          <EuiFlexItem grow={false}>
+            {first(values)}
+            {' ... '}
+          </EuiFlexItem>
+          <EuiFlexItem>
+            <EuiLink onClick={expand}>
+              <FormattedMessage
+                id="xpack.infra.nodeDetails.tabs.metadata.seeMore"
+                defaultMessage="+{count} more"
+                values={{
+                  count: values.length,
+                }}
+              />
+            </EuiLink>
+          </EuiFlexItem>
+        </EuiFlexGroup>
+      )}
+      {isExpanded && (
+        <div>
+          {values.map((v) => (
+            <div key={v}>{v}</div>
+          ))}
+          <EuiLink onClick={collapse}>
+            {i18n.translate('xpack.infra.nodeDetails.tabs.metadata.seeLess', {
+              defaultMessage: 'Show less',
+            })}
+          </EuiLink>
+        </div>
+      )}
+    </>
+  );
```

Comment:
Uploading show-more.mov…

As you can see in the following video, this part always renders the `Show more` button even if there is a single element in the list.

To delegate the logic to this list and also avoid checks on the column render method, you could refactor the component to:
1. Receive the value and transform it into an array if necessary
```js
const list = Array.isArray(value) ? value : [value]
```
2. Destructure the first item and group all the others with the rest operator (you get rid of lodash `first`)
```js
const [first, ...others] = list
```
3. Always render the first, and conditionally render all the others using the FlexItem if `others` contains more elements.
```jsx
const hasOthers = others.length > 0;
const shouldShowMore = hasOthers && !isExpanded;

return (
  <EuiFlexGroup
    gutterSize={'xs'}
    responsive={false}
    alignItems={'baseline'}
    wrap={true}
    direction="column"
  >
    <div>
      {first}
      {shouldShowMore && (
        <>
          {' ... '}
          <EuiLink onClick={expand}>
            <FormattedMessage
              id="xpack.infra.nodeDetails.tabs.metadata.seeMore"
              defaultMessage="+{count} more"
              values={{
                count: others.length,
              }}
            />
          </EuiLink>
        </>
      )}
    </div>
    {isExpanded && others.map((item) => <EuiFlexItem key={item}>{item}</EuiFlexItem>)}
    {hasOthers && isExpanded && (
      <EuiFlexItem>
        <EuiLink onClick={collapse}>
          {i18n.translate('xpack.infra.nodeDetails.tabs.metadata.seeLess', {
            defaultMessage: 'Show less',
          })}
        </EuiLink>
      </EuiFlexItem>
    )}
  </EuiFlexGroup>
)
```

Final result should be something like this

https://user-images.githubusercontent.com/34506779/224303590-d45126b5-4c2a-441a-89dc-5e9c4df7953b.mov

Reply:
Thank you for spotting that! 🙂  It's the same in inventory and I didn't pay too much attention to it but it's also a bug there (sorry, my bad :/ ). I will fix this one here and I leaving this as a note to add an issue to fix that in inventory as well (as I am trying to keep this PR focused and easier to be reviewed as it's already quite big 😅 )

Reply:
Added an [issue](https://github.com/elastic/kibana/issues/153373) to fix that in Inventory